RocketChat/Rocket.Chat
# [NEW] Prune messages support for new events architecture

## Proposed changes

This Pull Request implements the required changes and additions in order to the Prune Messages functionality be able to work properly.

## Issue(s)

This PR is related to https://github.com/RocketChat/Rocket.Chat/pull/15711 and it's base branch is the the one being developed on the related PR.

## How to test or reproduce

For testing this functionality you should have admin access and click on the 3 dots icon at the top right of the RC workspace > select **Prune Messages** > Apply any filter you wish the it should reflect in the room messages removing them.

## Screenshots

## Types of changes




## Checklist



## Changelog



n/a

## Further comments


### Important notes on this development

We had talked in the past about using the model of `Messages` for having backwards compatibility and the decisions of incrementing logic at the `RoomEvents` instead of `Messages` were applied for some reasons:
- Not only RoomEvents be the right model but also guaranteeing compatibility on the function signature is satisfactory for the Front-End <-> Back-End interoperability in this case;
- The methods used at `Messages` model were only used by the `cleanRoomHistory` function and have no impact in not using them;
- Keeping too specific methods when the problem is about building a query doesn't make sense, considering interface segregation principle as axiom;
- It would be necessary to work on how the methods in `Messages` model communicate data and it would create modification on the methods instead of the class be open for extension but closed for modification and it would raise architecture not so good practices and accidental complexity.
